### DISCUSSION TITLE
Take action based on multiple notification devices

### DISCUSSION THREAD

**Discussion by fabnavigator** [zwave-js/node-red-contrib-zwave-js#254] [Q&A]
I have a three door sensors that send out notification events (command class 113). I would like to take action when I get an event from one of these sensors, but the action I take is based on the state of all three sensors. The only thing I can think of is to maintain context for all three sensors and then read the context in a function node to take the desired action. Is this the way to do this, or is there something easier?

**Comment by marcus-j-davies** [comment]
Use a node such as Boolean logic.

https://flows.nodered.org/node/node-red-contrib-boolean-logic-ultimate

You can emit a true or false, based upon multiple (differently timed) events.

I use it in such a way that one of my sensors must be true (OR) but you can use it in AND mode (all sensors must be a certain value)

You just have to program your logic around it - it works extremely well.

It uses topics to  identify inputs (you must pass true or false for each topic) - can be at different times.

In OR mode - one must be true, to return true, else it's false 

In AND mode - all Must pass true to be true else false.

Then you can react on the overall output (true or false)

This node is designed to exactly cator for your needs

**Reply by fabnavigator** [reply]
@marcus-j-davies Thank you. I see how this node can be useful. I will probably just use context because the elapsed time since the last change is also a criteria for the decision. I don't think this node can do that.
